sw/device/tests/rv_core_ibex_nmi_irq_test.c

```diff
@@ -0,0 +1,263 @@
+ * but at the phase 1 (i.e. wipe secrets)the nmi interrupt handler should clear
+ * the escalation.
+ */
+enum {
+  kEscalationPhase0Micros = 100 * 1000,  // 10 ms
+  kEscalationPhase2Micros = 100,         // 100 us
+};
+
+static void alert_handler_config(void) {
+  dif_alert_handler_alert_t alerts[] = {kTopEarlgreyAlertIdPwrmgrAonFatalFault};
+  dif_alert_handler_class_t alert_classes[] = {kDifAlertHandlerClassA};
+
+  dif_alert_handler_escalation_phase_t esc_phases[] = {
+      {.phase = kDifAlertHandlerClassStatePhase0,
+       .signal = 0,
+       .duration_cycles = udiv64_slow(
+           kEscalationPhase0Micros * kClockFreqPeripheralHz, 1000000, NULL)},
+      {.phase = kDifAlertHandlerClassStatePhase2,
+       .signal = 3,
+       .duration_cycles = udiv64_slow(
+           kEscalationPhase2Micros * kClockFreqPeripheralHz, 1000000, NULL)}};
+
+  dif_alert_handler_class_config_t class_config[] = {{
+      .auto_lock_accumulation_counter = kDifToggleDisabled,
+      .accumulator_threshold = 0,
+      .irq_deadline_cycles =
+          udiv64_slow(10 * kClockFreqPeripheralHz, 1000000, NULL),
+      .escalation_phases = esc_phases,
+      .escalation_phases_len = ARRAYSIZE(esc_phases),
+      .crashdump_escalation_phase = kDifAlertHandlerClassStatePhase3,
+  }};
+
+  dif_alert_handler_class_t classes[] = {kDifAlertHandlerClassA};
+  dif_alert_handler_config_t config = {
+      .alerts = alerts,
+      .alert_classes = alert_classes,
+      .alerts_len = ARRAYSIZE(alerts),
+      .classes = classes,
+      .class_configs = class_config,
+      .classes_len = ARRAYSIZE(class_config),
+      .ping_timeout = 0,
+  };
+
+  alert_handler_testutils_configure_all(&alert_handler, config,
+                                        /*lock=*/kDifToggleDisabled);
+
+  // Enables alert handler irq.
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassa, kDifToggleEnabled));
+}
+
+/**
+ * Handle the wdog bark nmi irq.
+ */
+static void wdog_handler(void) {
+  top_earlgrey_plic_peripheral_t peripheral;
+  dif_aon_timer_irq_t irq;
+  isr_testutils_aon_timer_isr(plic_ctx, aon_timer_ctx, &peripheral, &irq);
+  CHECK_DIF_OK(dif_aon_timer_watchdog_stop(&aon_timer));
+}
+
+/**
+ * Handle the alert handler nmi irq.
+ */
+static void alert_handler_handler(void) {
+  top_earlgrey_plic_peripheral_t peripheral;
+  dif_alert_handler_irq_t irq;
+
+  isr_testutils_alert_handler_isr(plic_ctx, alert_handler_ctx, &peripheral,
+                                  &irq);
+
+  bool can_clear;
+  CHECK_DIF_OK(dif_alert_handler_escalation_can_clear(
+      alert_handler_ctx.alert_handler, kDifAlertHandlerClassA, &can_clear));
+
+  CHECK(can_clear, "Alert handler is locked.");
+  CHECK_DIF_OK(dif_alert_handler_escalation_clear(
+      alert_handler_ctx.alert_handler, kDifAlertHandlerClassA));
+}
+
+/**
+ * Execute the wdog bark nmi interrupt test.
+ */
+static void execute_wdog_nmi_test(void) {
```

Comment:
nit: Just so that this looks more similar to general test case conventions, what do you think about changing the name as follows:

s/`execute_wdog_nmi_test`/`wdog_nmi_test`/

Reply:
Looks better indeed, Fixed